istresearch/mailroom
# P4-1552 - Adjust for es 7, exit if ES is unavailable on start

# For the Reviewer

When this is complete, you should approve the PR via github.

## Summary
Adjusted mailroom so it is now ES7 compatible. Additionally added a return (rather than log) if es is unavailable on startup. 

#### Release Note
Mailroom is now ES7 compatible, and will exit if ES is unavailable on boot. 

#### Breaking Changes
Engage's contact database must be upgraded to ES 7. 

## Quality Assurance

You have gathered the following items:

Given all of the items above, you have updated your Application ORT at the following locations:
- **Features and Alerting**: <link to app ORT>Required.
- **P4 Alerting**: <link to p4-alerting PR>Required.

## Testing and Verification

- Create an Engage 4.0.3 stack with ES 7
- Ensure you can create, export, and search contacts.
- Ensure messages and flows continue to function.

To test the new restart functionality - start mailroom before elasticsearch on a local stack. Mailroom will exit and restart if docker is setup with a policy. When ES is started and green, mailroom will connect and stay up.
